File: scala/reverse-string/src/main/scala/ReverseString.scala
object ReverseString:
  // def reverse(str: String): String = str.reverse
  def reverse(str: String): String =
    str.foldRight("")((char, acc) => acc + char)

  def reverseOperation(chars: List[Char], acc: String): String =
    chars match
      case Nil          => acc
      case head :: tail => reverseOperation(tail, head + acc)

File: java/reverse-string/src/main/java/ReverseString.java
class ReverseString {

  String reverse(String inputString) {

    int i = 0;
    int j = inputString.length() - 1;
    char[] charArr = inputString.toCharArray();
    while (i <= j) {
      char temp = charArr[i];
      charArr[i] = charArr[j];
      charArr[j] = temp;
      i++;
      j--;
    }

    return new String(charArr);
  }
}
